File: jan-2014/fgfs/f.py
import sys

tests = int(raw_input())

for test in xrange(tests):
	nk = map(int, raw_input().split())
	n = nk[0]
	k = nk[1]

	spots = {}

	for nn in xrange(n):
		sfp = map(int, raw_input().split())
		s = sfp[0]
		f = sfp[1]
		p = sfp[2]

		if not p in spots:
			spots[p] = []

		spots[p].append((s, f))

	total = 0

	for spot in spots:
		ss = sorted(spots[spot], key=lambda x: x[1])
		O = []
		finish = 0
		for s in ss:
			if finish <= s[0]:
				finish = s[1]
				O.append(s)

		total += len(O)

	print total

File: jan-2014/fgfs/f.cpp
#include<iostream>
#include<stdio.h>
#include<tuple>
#include<vector>
#include<algorithm>
using namespace std;

typedef tuple<unsigned long,unsigned long> interval;

bool mycompare (const interval &lhs, const interval &rhs){
  return get<1>(lhs) < get<1>(rhs);
}

int main() {
	int tests;
	unsigned long n,k,s,f,p;
	scanf("%d", &tests);

	for (int i = 0; i < tests; i++) {
		scanf("%lu", &n);
		scanf("%lu", &k);

		vector<interval> spots[k];
/*
		for (long long j = 0; j < n; j++) {
			scanf("%lld", &s);
			scanf("%lld", &f);
			scanf("%lld", &p);

			spots[p-1].push_back(make_tuple(s,f));
		}

		long long total = 0;
		
		for (long long j = 0; j < k; j++) {
			sort(spots[j].begin(), spots[j].end(), mycompare);

			long long finish = 0;
			
			for (vector<interval>::iterator iter = spots[j].begin(); iter != spots[j].end(); iter++) {
				if (finish <= get<0>(*iter)) {
					finish = get<1>(*iter);
					total += 1;
				}
				//cout << get<0>(*iter) << "\t" << get<1>(*iter) << endl;
  			}
		}

		printf("%lld\n", total);
*/
	}
}
